webpack.config.js

```diff
@@ -45,6 +22,27 @@ function readCanisterId() {
   return canisterId;
 }
 
+// This plugin emulates the behaviour of http.rs while using webpack dev server locally
+// so we don't need to proxy to the backend canister for the index.html file.
+// This overcomes some issues with Safari and the CSP headers set in http.rs.
+class InjectCanisterIdPlugin {
```

Comment:
This could also be handled by using HtmlWebpackPlugin's templating features, but that means we would either have one messy index.html or maintain two separate ones. I think it's nicer to keep this complexity in the webpack config. But I'm happy to take the alternative approach if anyone disagrees.